std/encoding/binary.ts

```diff
@@ -35,34 +35,36 @@ const rawTypeSizes = {
   uint64: 8,
   float32: 4,
   float64: 8,
-};
+} as const;
 
-/** Returns the number of bytes required to store the given data-type. */
-export function sizeof(dataType: RawTypes): number {
+/** Number of bytes required to store `dataType`. */
+export function sizeof(dataType: DataType): number {
   return rawTypeSizes[dataType];
 }
 
 /** Reads `n` bytes from `r`.
  *
- * Returns it in a `Uint8Array`, or throws `Deno.errors.UnexpectedEof` if `n` bytes cannot be read. */
+ * Returns it in a `Uint8Array`, or null if `n` bytes cannot be read. */
 export async function getNBytes(
   r: Deno.Reader,
   n: number
-): Promise<Uint8Array> {
+): Promise<Uint8Array | null> {
   const scratch = new Uint8Array(n);
   const nRead = await r.read(scratch);
-  if (nRead === null || nRead < n) throw new Deno.errors.UnexpectedEof();
+  if (nRead === null || nRead < n) return null;
   return scratch;
 }
 
-/** Decode a number from `b`, and return it as a `number`. Data-type defaults to `int32`.
- * Returns `null` if `b` is too short for the data-type given in `o`. */
-export function varnum(b: Uint8Array, o: VarnumOptions = {}): number | null {
-  o.dataType = o.dataType ?? "int32";
+/** Decode a `number` from `o.bytes`.
+ *
+ * If `o.bytes` is shorter than `sizeof(o.type)`, returns `null`. */
+export function varnum(
+  o: VarnumOptions & { bytes: Uint8Array }
+): number | null {
```

Comment:
Why change the signature? I'm not sure it's an improvement.

Reply:
I thought changing `getNBytes` to `null` better follows Deno.read(), but really it should error because the user is specifying they know how many bytes there are to be read.
I also thought that the data-type should be compulsory, but it would make sense that that is created using the Options type first and used throughout the code.
I'll revert the changes to the signature, keep the new varnumBytes() methods (less messy sometimes than putVarnum), and update the comments to reflect a default. (Thanks for reviewing and keeping me in check :) )